libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt

```diff
@@ -29,28 +29,27 @@ data class NavigationOptions constructor(
     val navigatorPollingDelay: Long,
     val distanceFormatter: DistanceFormatter?,
     val onboardRouterConfig: MapboxOnboardRouterConfig?,
-    val isFromNavigationUi: Boolean = false
+    val isFromNavigationUi: Boolean
 ) {
 
     /**
      * Get a builder to customize a subset of current options.
      */
-    fun toBuilder() = Builder(
-        roundingIncrement,
-        timeFormatType,
-        navigatorPollingDelay,
-        distanceFormatter,
-        onboardRouterConfig
-    )
+    fun toBuilder() = Builder()
+        .roundingIncrement(roundingIncrement)
+        .timeFormatType(timeFormatType)
+        .navigatorPollingDelay(navigatorPollingDelay)
+        .distanceFormatter(distanceFormatter)
+        .onboardRouterConfig(onboardRouterConfig)
+        .isFromNavigationUi(isFromNavigationUi)
 
-    data class Builder(
```

Comment:
This change has removed the `equals` that comes with a data class from the builder though, which might impact unit tests downstream. We should either use a data class or implement equals and hash functions.

Reply:
Great catch! For reference 👀 https://www.baeldung.com/kotlin-builder-pattern cc @kmadsen

Reply:
Hmm maybe make a failure test @LukasPaczos ?

I tried and it works https://github.com/mapbox/mapbox-navigation-android/pull/2606

Reply:
Hmm but I changed it back to a data class and it also fails.

So that test fails with both implementations